miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/page/FormPage.java

```diff
@@ -0,0 +1,28 @@
+package uk.ac.bbsrc.tgac.miso.webapp.integrationtest.page;
+
+import java.util.Map;
+
+import org.openqa.selenium.WebDriver;
+
+/**
+ * Page that is mainly composed of a list of fields
+ * 
+ * @param <T> field enum
+ */
+public abstract class FormPage<T> extends HeaderFooterPage {
+
+  public FormPage(WebDriver driver) {
+    super(driver);
+  }
+
+  public abstract boolean isEditable(T field);
```

Comment:
You can make this simpler. Erase the generic, then add this interface:
```
public interface FieldElement {
public By getSelect();
public FieldType getType();
}
```
And make the enums extends it.
`public void setFields(Map<T, String> fields)`  can be changed to `public void setFields(Map<? extends FieldValue, String> fields)`

Then you could push all the logic you have in the derived classes and enums into this base class.